pkg/cmd/smithcmp/main.go

```diff
@@ -159,7 +164,13 @@ func main() {
 
 	var prep, exec string
 	ctx := context.Background()
+	done := time.After(timeout)
 	for i := 0; true; i++ {
+		select {
+		case <-done:
+			os.Exit(0)
```

Comment:
Can this get changed to a `return`? Could help if one day we have defers somewhere or this is moved into a helper function and we want some cleanup.